src/openms/include/OpenMS/ANALYSIS/ID/AScore.h

```diff
@@ -78,31 +78,81 @@ class OPENMS_DLLAPI AScore
 
         @note the original sequence is saved in the PeptideHits as MetaValue Search_engine_sequence.
     */
-    PeptideHit compute(PeptideHit & hit, RichPeakSpectrum &real_spectrum, double fragment_mass_tolerance, bool fragment_mass_unit_ppm) const;
-
+    PeptideHit compute(const PeptideHit & hit, RichPeakSpectrum &real_spectrum, double fragment_mass_tolerance, bool fragment_mass_unit_ppm) const;
+
+  protected:
+  
+    /// getSpectrumDifference_ works similar as the method std::set_difference (http://en.cppreference.com/w/cpp/algorithm/set_difference). 
+    /// set_difference was reimplemented, because it was necessary to overwrite the compare operator to be able to compare the m/z values.
```

Comment:
I'm not sure if that would work in every case. As far as I know, the comparison operator ("<") for sets etc. has to impose a total ordering on the domain. However, if you compare with tolerances, that isn't the case, which might lead to contradictions.

For example, if "<" compares with tolerances and you have two elements "a" and "b" that are within the tolerance of each other, then "a < b" and "b < a" are both false. If you now consider an element "c" that is within the tolerance of "b", but not "a", then you start to run into problems.